section-05-production-model-package/regression_model/processing/validation.py

```diff
@@ -29,9 +29,10 @@ def validate_inputs(*, input_data: pd.DataFrame) -> Tuple[pd.DataFrame, Optional
 
     # convert syntax error field names (beginning with numbers)
     input_data.rename(columns=config.model_config.variables_to_rename, inplace=True)
-    input_data.drop("Id", axis=1, inplace=True)
-    input_data["MSSubClass"] = input_data["MSSubClass"].astype("O")
-    validated_data = drop_na_inputs(input_data=input_data)
+    relevant_data = input_data[config.model_config.features].copy()
+    # TODO: Is this line redundant?
```

Comment:
In my notebooks, pandas automatically cast this var as numeric, and then I get an error from feature engine, because feature engine does not allow numerical variables in its categorical imputers / encoders.

Remove the line and see if you get the error. If not, then we are safe to remove.

Reply:
Runs without error, so I have removed